src/containers/Generator/Generator.js

```diff
@@ -0,0 +1,95 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux';
+import { setActivePalette } from '../../actions';
+import PropTypes from 'prop-types';
+import '../../main.scss';
+
+class Generator extends Component {
+  constructor() {
+    super()
+    this.state = {
+      locked: [],
+    }
+  }
+
+  componentDidMount() {
+    this.generatePalette()
+  }
+  
+  createColor = () => {
+    const options = '0123456789ABCDEF'
+    let hexArray = ['#']
+    for (let i = 0; i < 6; i++) {
+      let random = Math.floor(Math.random() * (16))
+      hexArray.push(options[random])
+    }
+    return hexArray.join('')
+  }
+
+  generatePalette = () => {
+    const { locked } = this.state
+    const {activePalette} = this.props
+    let palette = []
+    for (let i = 0; i < 5; i++) {
+      let color = ''
+      if (locked.includes(i)) {
+        color = activePalette[i]
+      } else {
+        color = this.createColor()
+      }
+      palette.push(color)
+    }
+    this.props.setActivePalette(palette)
+  }
+
+  handleClick = (e) => {
+    const { value } = e.target
+    const { locked } = this.state
+    let checked = []
+    if (locked.includes(parseInt(value))) {
+      checked = locked.filter(id => id !== parseInt(value))
+    } else {
+      checked = [...locked, parseInt(value)]
+    }
```

Comment:
This is a bit weird, we should change this to something else like toggleLock
it makes sense but I think we could consider refactoring this a bit. Lets discuss this in the afternoon